docs/tutorial/flink.md

```diff
@@ -1,94 +1,407 @@
+
+   recipe = (
+       beam.Create(pattern.items())
+       | OpenURLWithFSSpec()
+       | OpenWithXarray(file_type=pattern.file_type, xarray_open_kwargs={"decode_coords": "all"})
+       | StoreToZarr(
+           store_name="gpcp",
+           combine_dims=pattern.combine_dim_keys,
+       )
+   )
+   ```
+
+### Where Data is Cached
+
+The inputs and metadata can be cached so items don't have to be resolved again during the next recipe run. Defining where
+the caching happens is part of the runner file configuration. More about that in the [next session](#configfiles). Note 
+that in those upcoming examples keys `InputCacheStorage`` target these options.
+
+### Where Data is Output
+
+Defining where the data is output is also part of the runner file configuration. More about that in the [next session](#configfiles). Note 
+that in those upcoming examples the key `TargetStorage` defines this option.
+
+
+<a name="configfiles"></a>
+
+### Configuration Files
+
+Let's build a configuration file for where data is cached and output. Below we show two different formats for this configuration file.
+
+Notice the use of `{{job_name}}` in the `root_path` configuration examples below.
+`{{job_name}}` is special within `root_path` and will be treated as a template-value based on
+`Bake.job_name` which can be provided through the CLI or in the configuration file itself and, failing
+to be provided, will be generated automatically.
+
+Also notice that we are going to store everything in s3 below because we chose `s3fs.S3FileSystem`. This isn't a requirement.
+Pangeo-Forge aims to be storage agnostic. By depending on `fsspec`  we're able to plug in supported backends.
+Review other well-known `fsspec` [built-in implementations](https://filesystem-spec.readthedocs.io/en/latest/api.html#built-in-implementations)
+and `fsspec` [third-party implementations](https://filesystem-spec.readthedocs.io/en/latest/api.html#other-known-implementations).
+
+1. JSON configuration:
+
+   ```json
+   {
+     "TargetStorage": {
+       "fsspec_class": "s3fs.S3FileSystem",
+       "fsspec_args": {
+          "key": "<your-aws-access-key>",
+          "secret": "<your-aws-access-secret>",
+          "client_kwargs":{"region_name":"<your-aws-bucket-region>"}
+        },
+       // Target output should be partitioned by `{{job_name}}`
+       "root_path": "s3://<bucket-name>/<some-prefix>/{{job_name}}/output"
+     },
+     "InputCacheStorage": {
+       "fsspec_class": "s3fs.S3FileSystem",
+         "fsspec_args": {
+          "key": "<your-aws-access-key>",
+          "secret": "<your-aws-access-secret>",
+          "client_kwargs":{"region_name":"<your-aws-bucket-region>"}
+        },
+       // Input data cache should *not* be partitioned by `{{job_name}}`, as we want to get the datafile from the source only once
+       "root_path": "s3://<bucket-name>/<some-prefix>/input/cache"
+     }
+   }
+   ```
+
+2. `traitlet` configuration:
```

Comment:
```suggestion
2. Python based configuration file
```

Both of these are using traitlets, regardless of the format of the configuration file.